src/omlt/base/constraint.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pyomo.environ as pyo
+from pyomo.core.expr import EqualityExpression, InequalityExpression
+
+from omlt.base import DEFAULT_MODELING_LANGUAGE
+
+
+class OmltConstraint:
+    @property
+    def ctype(self):
+        return pyo.Constraint
+
+    def valid_model_component(self):
+        """Return True if this can be used as a model component."""
+        return True
+
+
+class OmltConstraintScalar(OmltConstraint):
+    def __init__(self, lang: str = DEFAULT_MODELING_LANGUAGE, **kwargs: Any):
```

Comment:
It would be helpful if there was some indication of what the kwargs could be here. See file comment on documenting abstract classes and methods.